airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJooqSource.java

```diff
@@ -0,0 +1,341 @@
+    final Instant now = Instant.now();
+
+    final Database database = createDatabase(config);
+
+    final Map<String, Table<?>> tableNameToTable = discoverInternal(database).stream()
+        .collect(Collectors.toMap(t -> String.format("%s.%s", t.getSchema().getName(), t.getName()), Function.identity()));
+
+    Stream<AirbyteMessage> resultStream = Stream.empty();
+
+    for (final ConfiguredAirbyteStream airbyteStream : catalog.getStreams()) {
+      final String streamName = airbyteStream.getStream().getName();
+      if (!tableNameToTable.containsKey(streamName)) {
+        continue;
+      }
+
+      final Set<String> selectedFields = CatalogHelpers.getTopLevelFieldNames(airbyteStream);
+
+      final Table<?> table = tableNameToTable.get(streamName);
+      final List<org.jooq.Field<?>> selectedDatabaseFields = Arrays.stream(table.fields())
+          .filter(field -> selectedFields.contains(field.getName()))
+          .collect(Collectors.toList());
+
+      if (selectedDatabaseFields.isEmpty()) {
+        continue;
+      }
+
+      final Stream<AirbyteMessage> stream;
+      if (airbyteStream.getSyncMode() == SyncMode.INCREMENTAL) {
+        final JdbcStateManager stateManager = new JdbcStateManager(state == null ? new JdbcState() : Jsons.object(state, JdbcState.class));
+        final String cursorField = IncrementalUtils.getCursorField(airbyteStream);
+        final JsonSchemaPrimitive cursorType = IncrementalUtils.getCursorType(airbyteStream, cursorField);
+        final Optional<String> initialCursorOptional = stateManager.getOriginalCursor(streamName);
+
+        final Stream<AirbyteMessage> internalMessageStream;
+        if (initialCursorOptional.isPresent()) {
+          final org.jooq.Field<?> cursorJooqField = Arrays
+              .stream(table.fields())
+              .filter(f -> f.getName().equals(cursorField))
+              .findFirst()
+              .orElseThrow(
+                  () -> new IllegalStateException(String.format("Could not find cursor field %s in table %s", cursorField, table.getName())));
+
+          final Stream<Record> queryStream = executeIncrementalQuery(
+              database,
+              selectedDatabaseFields,
+              table,
+              cursorJooqField,
+              initialCursorOptional.get());
+          internalMessageStream = getMessageStream(queryStream, streamName, now.toEpochMilli());
+        } else {
+          internalMessageStream = getMessageStream(executeFullRefreshQuery(database, selectedDatabaseFields, table), streamName, now.toEpochMilli());
+        }
+
+        final StateDecoratingIterator stateDecoratingIterator = new StateDecoratingIterator(
+            internalMessageStream,
+            stateManager,
+            streamName,
+            cursorField,
+            initialCursorOptional.orElse(null),
+            cursorType);
+
+        stream = MoreStreams.toStream(stateDecoratingIterator);
```

Comment:
rename to stateStream?

Reply:
that's not accurate because this variable gets reused by the full refresh branch.